tool/python/singa/layer.py

```diff
@@ -0,0 +1,431 @@
+#* with the License.  You may obtain a copy of the License at
+#*
+#*   http://www.apache.org/licenses/LICENSE-2.0
+#*
+#* Unless required by applicable law or agreed to in writing,
+#* software distributed under the License is distributed on an
+#* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#* KIND, either express or implied.  See the License for the
+#* specific language governing permissions and limitations
+#* under the License.
+#*
+#*************************************************************/
+
+'''
+This script includes Layer class and its subclasses that
+users can configure different types of layers for their model.
+'''
+
+from singa.parameter import Parameter, set_param_field
+from singa.initializations import get_init_values
+from singa.utils.utility import setval, generate_name
+from singa.utils.message import *
+from google.protobuf import text_format
+
+class Layer(object):
+
+    def __init__(self, **kwargs):
+        '''
+        **kwargs (KEY=VALUE)
+          partition_dim = (int)  // partition dimension for net
+        '''
+
+        self.layer = Message('Layer', **kwargs).proto
+        # required
+        if not 'name' in kwargs:
+            setval(self.layer, name=generate_name('layer', 1))
+
+        # srclayers are set in Model.build()
+        self.is_datalayer = False
+
+class Data(Layer):
+
+    def __init__(self, load, phase='train', checkpoint=None,
+                 conf=None, **kwargs):
+        '''
+        required
+          load       = (string)  // type of data
+        optional
+          phase      = (string)  // phase of data layer
+          checkpoint = (string)  // checkpoint path
+          conf       = (Store)   // Store object
+          **kwargs (KEY=VALUE)
+            partition_dim = (int)  // partition dimension for net
+        '''
+
+        assert load != None, 'data type should be specified'
+        if load == 'kData':
+            super(Data, self).__init__(name=generate_name('data'),
+                                       user_type=load)
+        else:
+            self.layer_type = enumLayerType(load)
+            super(Data, self).__init__(name=generate_name('data'),
+                                       type=self.layer_type)
+        self.is_datalayer = True
+
+        # include/exclude
+        setval(self.layer, include=enumPhase(phase))
+        #setval(self.layer, exclude=kTest if phase=='train' else kTrain)
+
+        if conf == None:
+            if load == 'kData':
+                setval(self.layer.Extensions[data_conf], **kwargs)
+            else:
+                setval(self.layer.store_conf, **kwargs)
+        else:
+            setval(self.layer, store_conf=conf.proto)
+
+        self.checkpoint = checkpoint # checkpoint for training data
+
+
+class Convolution2D(Layer):
+
+    def __init__(self, nb_filter=0, kernel=0, stride=1, pad=0,
+                 init=None, w_param=None, b_param=None,
+                 activation=None, **kwargs):
```

Comment:
the w_std, and b_lr used in the cifar10 example are not defined here?
users may not know how to use these two arguments.
it would be better to put them in the argument list.